app/code/Codifi/CustomerRequest/Model/ExportCsv.php

```diff
@@ -0,0 +1,175 @@
+     * TimezoneInterface
+     *
+     * @var TimezoneInterface
+     */
+    private $timeZoneInterface;
+
+    /**
+     * New directory
+     */
+    private $newDirectory;
+
+    /**
+     * ExportCsv constructor.
+     *
+     * @param Filesystem $fileSystem
+     * @param NoteRepository $noteRepository
+     * @param Csv $csvProcessor
+     * @param DirectoryList $directoryList
+     * @param FilterBuilder $filterBuilder
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param TimezoneInterface $timeZoneInterface
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        NoteRepository $noteRepository,
+        Csv $csvProcessor,
+        DirectoryList $directoryList,
+        FilterBuilder $filterBuilder,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        TimezoneInterface $timeZoneInterface
+    ) {
+        $this->noteRepository = $noteRepository;
+        $this->csvProcessor = $csvProcessor;
+        $this->directoryList = $directoryList;
+        $this->filterBuilder = $filterBuilder;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->timeZoneInterface = $timeZoneInterface->date();
+        $this->newDirectory = $fileSystem->getDirectoryWrite(DirectoryList::VAR_DIR);
+    }
+
+    /**
+     * Pack customer notes to archive and delete from table
+     *
+     * @param int $period
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    public function export(int $period): string
+    {
+        $currentDateTime = $this->timeZoneInterface->format('Y_m_d H:m:s');
+
+        $date = strtotime($currentDateTime . "-1 $period");
+        $newDate = date('Y-m-d H:m:s', $date);
+
+        $updatedAt = $newDate;
+
+        $this->filterBuilder->setField('updated_at');
+        $this->filterBuilder->setConditionType('to');
+        $this->filterBuilder->setValue($updatedAt);
+        $filter = $this->filterBuilder->create();
+
+        $this->searchCriteriaBuilder->addFilter($filter);
+        $searchCriteria = $this->searchCriteriaBuilder->create();
+
+        $noteList = $this->noteRepository->getList($searchCriteria);
+
+        $noteListItems = $noteList->getItems();
+
+        $content[] = [
+            'note_id' => __('Note ID'),
+            'customer_id' => __('Customer ID'),
+            'created_at' => __('Created At'),
+            'created_by' => __('Created By'),
+            'note' => __('Note'),
+            'updated_at' => __('Updated At'),
+            'updated_by' => __('Updated By'),
+            'autocomplete' => __('Autocomplete')
+        ];
+
+        foreach ($noteListItems as $item) {
+            $note = $item->getData();
+            $content[] = $note;
+            $this->noteRepository->deleteById($note['note_id']);
+        }
+
+        try {
+            $currentDateForName = $this->timeZoneInterface->format('Y_m_d');
+
+            $this->newDirectory->create('/archive/');
```

Comment:
You should check that directory is created successfully and directory is writable